private/pkg/protosource/option_extension_descriptor.go

```diff
@@ -39,9 +39,23 @@ func (o *optionExtensionDescriptor) OptionExtension(extensionType protoreflect.E
 	if extensionType.TypeDescriptor().ContainingMessage().FullName() != o.message.ProtoReflect().Descriptor().FullName() {
 		return nil, false
 	}
+
 	if !proto.HasExtension(o.message, extensionType) {
 		return nil, false
 	}
+
+	if extensionType.TypeDescriptor().Kind() == protoreflect.MessageKind {
```

Comment:
Can you explain (and in comments too) what is going on here/the motivation here a bit more?

Reply:
sure, somewhat of a stop-gap solution to prevent dynamic type mismatching panics. Before the PR is ready i will revisit this.